packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -103,58 +106,66 @@ export class JobService {
     requestType: JobRequestType,
     dto: JobFortuneDto | JobImageLabelBinaryDto,
   ): Promise<number> {
-    const {
-      chainId,
-      submissionsRequired,
-      requesterDescription,
-      fundAmount,
-    } = dto;
+    let manifestUrl, manifestHash;
+    const { chainId, fundAmount } = dto;
 
     if (chainId) {
       this.web3Service.validateChainId(chainId);
     }
 
     const userBalance = await this.paymentService.getUserBalance(userId);
 
-    const fundAmountInWei = ethers.utils.parseUnits(
-      fundAmount.toString(),
-      'ether',
-    );
-
     const rate = await getRate(Currency.USD, TokenId.HMT);
 
-    const jobLauncherFee = BigNumber.from(
-      this.configService.get<number>(ConfigNames.JOB_LAUNCHER_FEE)!,
-    )
-      .div(100)
-      .mul(fundAmountInWei);
+    const feePercentage = this.configService.get<number>(
+      ConfigNames.JOB_LAUNCHER_FEE,
+    )!;
+    const fee = mul(div(feePercentage, 100), fundAmount);
+    const usdTotalAmount = add(fundAmount, fee);
 
-    const usdTotalAmount = BigNumber.from(
-      FixedNumber.from(
-        ethers.utils.formatUnits(fundAmountInWei.add(jobLauncherFee), 'ether'),
-      ).mulUnsafe(FixedNumber.from(rate.toString())),
-    );
-
-    if (userBalance.lt(usdTotalAmount)) {
+    if (lt(userBalance, usdTotalAmount)) {
       this.logger.log(ErrorJob.NotEnoughFunds, JobService.name);
       throw new BadRequestException(ErrorJob.NotEnoughFunds);
     }
 
-    const { manifestUrl, manifestHash } = await this.saveManifest({
-      ...dto,
-      requestType,
-      submissionsRequired,
-      requesterDescription,
-      fundAmount: fundAmountInWei.toString(),
-    });
+    const tokenFundAmount = mul(fundAmount, rate);
+    const tokenFee = mul(fee, rate);
+    const tokenTotalAmount = add(tokenFundAmount, tokenFee);
+
+    if (requestType == JobRequestType.FORTUNE) {
+      ({ manifestUrl, manifestHash } = await this.saveManifest({
+        ...(dto as JobFortuneDto),
+        requestType,
+        fundAmount: tokenFundAmount,
+      }));
+    } else {
+      dto = dto as JobImageLabelBinaryDto;
```

Comment:
I believe we need to rename `JobImageLabelBinaryDto` because this manifest will be used for `IMAGE_POINTS` and `IMAGE_BOXES`.